electrum/plugins/swapserver/__init__.py

```diff
@@ -1,5 +1,91 @@
+from typing import TYPE_CHECKING, List
+
 from electrum.simple_config import ConfigVar, SimpleConfig
+from electrum.commands import plugin_command
+
+if TYPE_CHECKING:
+    from electrum.commands import Commands
+    from electrum.wallet import Abstract_Wallet
+
+
+plugin_name = "swapserver"
+
 
 SimpleConfig.SWAPSERVER_PORT = ConfigVar('plugins.swapserver.port', default=None, type_=int, plugin=__name__)
 SimpleConfig.SWAPSERVER_FEE_MILLIONTHS = ConfigVar('plugins.swapserver.fee_millionths', default=5000, type_=int, plugin=__name__)
 SimpleConfig.SWAPSERVER_ANN_POW_NONCE = ConfigVar('plugins.swapserver.ann_pow_nonce', default=0, type_=int, plugin=__name__)
+
+
+@plugin_command('wl', plugin_name)
+async def get_history(self: 'Commands', wallet: 'Abstract_Wallet' = None, plugin = None) -> List[dict]:
+    """
+Get a list of all swaps provided by this swapserver.
+Single elements can potentially cover multiple swaps if transactions have been batched.
+
+    Example result:
```

Comment:
I think docstrings should be indented so that they don't use the top/outermost indent.
e.g.
```suggestion
    """
    Get a list of all swaps provided by this swapserver.
    Single elements can potentially cover multiple swaps if transactions have been batched.

    Example result:
```
The only reason you would want to do this kind of indenting, is if the string is sensitive to whitespaces. E.g. consider: https://github.com/spesmilo/electrum/blob/8abe4a1dd44727cae532eadcec97cf72c1bb2cb4/electrum/simple_config.py#L691-L697
There the whitespaces would end up being in the string, and displayed in the GUI in an ugly way.

Reply:
The reason i did this was because it seems like the only option to get the first two lines of the docstring to align. Otherwise the second line is intended while the first one isn't. 
However this behavior is similar with other commands too so i just changed it now according to your suggestion.
```
usage: run_electrum swapserver_get_summary [-h]

Get a summary of all swaps provided by this swapserver.
    Can become incorrect if closed lightning channels have been deleted in this wallet.

    Example result:
    {
        "num_swaps": 160,
        "overall_return_sat": 159052,  # value earned or lost in satoshi
        "swaps_per_day": 0.78  # between first swap and last swap
    }

options:
  -h, --help  show this help message and exit

Run 'electrum -h' to see the list of global options
```

Reply:
Weird. That's not how it looks like for me:
```
$ ./run_electrum --regtest --dir /tmp/bob swapserver_get_summary -h
usage: run_electrum swapserver_get_summary [-h]

Get a summary of all swaps provided by this swapserver.
Can become incorrect if closed lightning channels have been deleted in this wallet.

Example result:
{
    "num_swaps": 160,
    "overall_return_sat": 159052,  # value earned or lost in satoshi
    "swaps_per_day": 0.78  # between first swap and last swap
}

options:
  -h, --help  show this help message and exit

Run 'electrum -h' to see the list of global options
```

Do you see this behaviour for all commands? E.g. I see:
```
$ ./run_electrum lnpay -h
usage: run_electrum lnpay [-h] [--timeout TIMEOUT] [--max_cltv MAX_CLTV] [--max_fee_msat MAX_FEE_MSAT]
                          [--password PASSWORD]
                          invoice

Pay a lightning invoice
Note: it is *not* safe to try paying the same invoice multiple times with a timeout.
      It is only safe to retry paying the same invoice if there are no more pending HTLCs
      with the same payment_hash.  # FIXME should there even be a default timeout? just block forever.

positional arguments:
  invoice               Lightning invoice (bolt 11)

options:
  -h, --help            show this help message and exit
  --timeout TIMEOUT     Timeout in seconds (default=120)
  --max_cltv MAX_CLTV   Maximum total time lock for the route (default=4032+invoice_final_cltv_delta)
  --max_fee_msat MAX_FEE_MSAT
                        Maximum absolute fee budget for the payment (if unset, the default is a percentage fee
                        based on config.LIGHTNING_PAYMENT_FEE_MAX_MILLIONTHS)
  --password PASSWORD   Wallet password. Use '--password :' if you want a prompt.

Run 'electrum -h' to see the list of global options
```

Reply:
I did yes, but now i cannot reproduce anymore. Maybe it was some terminal bug.